File: front/src/models/Product.ts
// models/Exp.ts
export interface Product {
  id: number;
  name: string;
  code: string;
  stock: number;
}

// 임의의 상품 데이터 생성
export const sampleProduct: Product[] = [
  {
    id: 1,
    name: '우유500ml',
    code: 'ML-500-01',
    stock: 6,
  },
  {
    id: 2,
    name: '샌드위치',
    code: 'SW-100-02',
    stock: 8,
  },
];

File: back/src/main/java/com/expmng/back/modules/product/infrastructure/entities/Product.java
package com.expmng.back.modules.product.infrastructure.entities;

import jakarta.persistence.Entity;
import jakarta.persistence.*;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.time.LocalDateTime;
import java.util.List;

import lombok.AllArgsConstructor;

@Entity
@Table(name = "products")
@Data
@Builder
@AllArgsConstructor
@NoArgsConstructor
public class Product {
	@Id
	@GeneratedValue(strategy = GenerationType.IDENTITY)
	private Long id;

	@Column(nullable = false)
	private String name;

	@Column(length = 100)
	private String code;

	@Column(nullable = false)
	private Integer price;

	private Integer stock;

	@OneToMany(mappedBy = "product", cascade = CascadeType.ALL, orphanRemoval = true)
	private List<Exp> exps;

	@Column(name = "created_at")
	private LocalDateTime createdAt;

	@Column(name = "updated_at")
	private LocalDateTime updatedAt;

	@PrePersist
	protected void onCreate() {
		createdAt = LocalDateTime.now();
		updatedAt = LocalDateTime.now();
	}

	@PreUpdate
	protected void onUpdate() {
		updatedAt = LocalDateTime.now();
	}
}